src/Products/Signature/Controllers/SignatureApiController.cs

```diff
@@ -179,53 +151,65 @@ public HttpResponseMessage LoadDocumentDescription(SignaturePostedDataEntity pos
+                            description.height = documentInfo.Pages[i].Height;
+                            description.width = documentInfo.Pages[i].Width;
+                            description.number = i + 1;
+                            if (GlobalConfiguration.Signature.PreloadPageCount == 0)
+                            {
+                                byte[] pageBytes = RenderPageToMemoryStream(signature, i, fileStream, password).ToArray();
+                                string encodedImage = Convert.ToBase64String(pageBytes);
+                                pageBytes = null;
+                                description.SetData(encodedImage);
+                            }
+                            pagesDescription.Add(description);
+                        }
+                        loadDocumentEntity.SetGuid(documentGuid);
+                        foreach (SignatureLoadedPageEntity pageDescription in pagesDescription)
                         {
-                            byte[] pageBytes = SignatureHandler.GetPageImage(stream, i, password, null, 100);
-                            string encodedImage = Convert.ToBase64String(pageBytes);
-                            pageBytes = null;
-                            description.SetData(encodedImage);
+                            loadDocumentEntity.SetPages(pageDescription);
                         }
-                        pagesDescription.Add(description);
-                    }
-                    loadDocumentEntity.SetGuid(documentGuid);
-                    foreach (SignatureLoadedPageEntity pageDescription in pagesDescription)
-                    {
-                        loadDocumentEntity.SetPages(pageDescription);
                     }
                 }
                 // return document description
                 return Request.CreateResponse(HttpStatusCode.OK, loadDocumentEntity);
             }
+            catch (PasswordRequiredException ex)
+            {
+                // set exception message
+                return Request.CreateResponse(HttpStatusCode.Forbidden, new Common.Resources.Resources().GenerateException(ex, password));
+            }
             catch (System.Exception ex)
             {
-                // TODO: this should be changed on special catch for PasswordProtectedException when it will be added in the library
-                if (ex.Message == "Invalid password")
-                {
-                    return Request.CreateResponse(HttpStatusCode.Forbidden, new Common.Resources.Resources().GenerateException(ex, password));
-                }
-
                 // set exception message
                 return Request.CreateResponse(HttpStatusCode.InternalServerError, new Common.Resources.Resources().GenerateException(ex, password));
             }
         }
 
+        static MemoryStream RenderPageToMemoryStream(GroupDocs.Signature.Signature signature, int pageNumberToRender, FileStream fileStream, string password)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'password'.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Signature-for-.NET-MVC/pullRequest?prid=5325147)